### ISSUE TITLE
Resolving foreign issues fails with with {options: {unique: true}}

### ISSUE THREAD

**Issue by janklan** [city41/node-sql-fixtures#48]
Hey there, thanks for this library!

I'm wondering if there is any known solution for the following.

**What's happening:**

1. My script is using `{options: {unique: true}}` to avoid inserting data repeatedly (I really love the idea of not having to wipe the database all the time!)
2. I've got a simple relation pointing at a parent row in the same table. See the code below.
3. I'm resolving the relation according to the documentation, using "property:0".
4. The first time I run my fixtures, everything works fine
5. The second time I get an error `Error: ER_TRUNCATED_WRONG_VALUE_FOR_FIELD: Incorrect integer value: 'property:__genned_specId_0' for column 'property_id_parent' at row 1` because the generated query is `insert into property (data_type, name, property_id_parent) values ('string', 'Model', 'property:__genned_specId_0')`

It appears that because (a) the row Brand already exists, (b) the relation description I'm using refers to the data in my JSON, not existing data in the database, (c) the `unique: true` prevents the "Brand" from being re-created", and (d) sql-fixtures fails to pick up that the Brand already exists in the database.

**What I expected to happen**

Two possible scenarios, actually:

1. The error should not have appeared in the first place. Both of the rows already exist in the database, I'm not quite sure why is sql-fixtures failing on creating one of them.
2. Ideally sql-fixtures would store the primary key of my "Brand" record when it decides the record already exists and should not be created. The second step is that the Model would be able to access the existing primary key value of Brand record. In saying so, the error in the Brand ID resolution occurred regardless of whether the Model row existed or not.

---

Here are my files, however, I suspect event the readme example would act the same.

**fixtures.js**
```javascript
var sqlFixtures = require('sql-fixtures');

var dbConfig = {
  client: 'mysql',
  connection: {
    host: 'localhost',
    user: 'root',
    password: 'root',
    database: 'db',
    port: 3306
  }
};

const property = require('./dev/property.json');

console.log(property);

var dataSpec = {
  property
};

var options = {
  unique: true,
  showWarning: false
};

sqlFixtures.create(dbConfig, dataSpec, options).then(result => {
  console.log(result);
  console.log('Done.');
  sqlFixtures.destroy();
}).catch(e => {
  console.log(e);
  sqlFixtures.destroy();
});
```

**./dev/property.json**
```json:
[
  {
    "name": "Brand",
    "data_type": "string"
  },
  {
    "name": "Model",
    "data_type": "string",
    "property_id_parent": "property:0"
  }
]
```

**Comment by city41** [comment]
Thanks for the report. I see why this is happening. The `unique` feature is pretty primitive. Before doing anything, all the rows in the spec are checked to see if they are already in the database. If they are, they get ignored, and so they can't participate in dependency resolution, so `property:0` ends up pointing at nothing.

`unique` is experimental and not the greatest because all it does is run a select with the where set to the record in the spec.

Let me think on this and see if I can find a clean way to fix this.

**Comment by city41** [comment]
For starters, wrote a test that repros the bug

https://github.com/city41/node-sql-fixtures/commit/3db826c0224107b7d43a6c9087a0198bf71244b5

**Comment by janklan** [comment]
Thanks for the speedy response!

I think it would make sense to go one stwp further with the checking: at the time you know whether a record exists or not, you also know what's the ID stored in the database. That's an opportunity to store that id so it can be used later, as if it was just inserted.

Does that sound reasonable?

**Comment by city41** [comment]
Yeah that seems like the basic approach to take. But need to hold onto the entire row, and also keep it's position in the spec. The code really wasn't written with this in mind, so I think it's going to take some refactoring and a bit of rethinking to get it in. Need to add more tests too, the one test I wrote is just the simplest case.


And just a heads up, not sure when I'll actually get around to implementing this, but I will make an effort.

**Comment by janklan** [comment]
Cheers mate! Share your thoughts when ready and I might be able to help with a PR.

**Comment by city41** [comment]
I think the basic approach is, inside `insert-records.js`

* remove `getInsertableRecords()`, who's job is to check if the records exist when unique is on
* in `insertRecordsSerially()`, first check if the record exists if unique is true, and if it does, use that row as the result for that record, otherwise insert it as before.
* in `buildInsertPromise()`, allow postgres to keep doing batch inserts when unique is false, but when it's true have postgres also go down the `insertRecordsSerially` path.

**Comment by city41** [comment]
Just fyi I added the help wanted label because I don't see myself getting to this any time soon. Once I get a some free cycles I will fix this. When I do take this on, I'll update here.  
  
If anyone else wants to take a stab, by all means.

**Comment by city41** [comment]
@janklan I just published `sql-fixtures@2.0.0-beta.0` which should fix this issue. Can you try it and see if it works for you?

Bumped to 2.0.0 because this changes how unique works.

here is the changed behavior
```
const dataSpec = {
    // this row already exists in the db
    myTable: {
        someColumn: 'my value'
    }
};

sqlFixtures(dataSpec, { unique: true }).then(results => {
    // for 1.x.x, results.myTable.length is zero

    // for 2.x.x, results.myTable.length is one, and the result is the existing row
});
```

**Comment by city41** [comment]
I am closing this due to deciding to no longer support this module. I have been mulling this over for a long time, and decided it was the right thing to do. For the record, this issue has nothing to do with my decision.
